x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportDeleteEnrichPolicyAction.java

```diff
@@ -53,7 +61,15 @@ protected AcknowledgedResponse read(StreamInput in) throws IOException {
     @Override
     protected void masterOperation(Task task, DeleteEnrichPolicyAction.Request request, ClusterState state,
                                    ActionListener<AcknowledgedResponse> listener) throws Exception {
+        try {
+            enrichPolicyLocks.lockPolicy(request.getName());
+        } catch (EsRejectedExecutionException e) {
+            listener.onFailure(new ElasticsearchStatusException("Could not delete policy [{}] because it is currently executing",
```

Comment:
Can we just throw the `EsRejectedExecutionException` that `lockPolicy (...)` threw?

Reply:
The only reason i didnt was because the error message was slightly different. It says `Policy execution failed. Policy execution for....` which is not exactly what is happening. Maybe we should change the error message in the lock to something like `Could not obtain lock because policy execution for [policyname] is in progress`

Reply:
see 340ae6a